KIPs/kip-dreaft_sdk_common_architecture.md

```diff
@@ -0,0 +1,1623 @@
+
+##### Variable description
+
+| Variable | Description |
+| ----------- | ----------- |
+| privateKey: String | The private key string. |
+
+##### Method description
+
+| Method | Description |
+| ----------- | ----------- |
+| sign(txHash: String, chainId: String): SignatureData | Signs the transaction. |
+| sign(txHash: String, chainId: int): SignatureData | Signs the transaction. |
+| signMessage(messageHash: String): SignatureData | Signs the message. |
+| getPublicKey(compressed: Boolean): String | Returns public key string. |
+| getDerivedAddress(): String | Returns derived address from private key string. |
+
+#### AbstractKeyring
+
+`AbstractKeyring` is an abstract class that abstracts Keyring classes that store the account address and private key(s) to use. All Keyring classes are implemented by extending `AbstractKeyring`.
+
+##### Variable description
+
+| Variable | Description |
+| ----------- | ----------- |
+| address: String | The address of the account. |
+
+##### Method description
+
+| Method | Description |
+| ----------- | ----------- |
+| sign(txHash: String, chainId: int, role: int): List&#60;SignatureData&#62; | Signs the transaction using key(s) defined in role. `sign` is defined as an "abstract method" in AbstractKeyring, and must be implemented in all keyring classes that extend `AbstractKeyring`. |
+| sign(txHash: String, chainId: int, role: int, index: int): SignatureData | Signs the transaction using a key defined in role. `sign` is defined as an "abstract method" in AbstractKeyring, and must be implemented in all keyring classes that extend `AbstractKeyring`. |
+| signMessage(message: String, role: int): MessageSigned | Signs the message using key(s) defined in role. `signMessage` is defined as an "abstract method" in AbstractKeyring, and must be implemented in all keyring classes that extend `AbstractKeyring`. |
+| signMessage(message: String, role: int, index: int): MessageSigned | Signs the message using a key defined in role. `signMessage` is defined as an "abstract method" in AbstractKeyring, and must be implemented in all keyring classes that extend `AbstractKeyring`. |
+| encrypt(password: String, options: Object): Object | Encrypts a Keyring instance with keystore v4 format. `encrypt` is defined as an "abstract method" in AbstractKeyring, and must be implemented in all keyring classes that extend `AbstractKeyring`. |
+| copy(): AbstractKeyring | Returns a copied Keyring instance. `copy` is defined as an "abstract method" in AbstractKeyring, and must be implemented in all keyring classes that extend `AbstractKeyring`. |
+| sign(txHash: String, chainId: String, role: int): List&#60;SignatureData&#62; | Signs the transaction using key(s) defined in role. Converts the type of chainId inputted as String into int and calls the sign function implemented in keyring. |
```

Comment:
```suggestion
| sign(txHash: String, chainId: String, role: int): List&#60;SignatureData&#62; | Signs the transaction using the key(s) specified by the role. It converts the type of chainId into int and calls `sign`. |
```

Reply:
@jimni1222 Is chainId converted to int or big int?

Reply:
chainId will be converted inside of the singing logic.